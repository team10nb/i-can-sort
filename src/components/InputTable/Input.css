.table{
    height: 40px;
    width: 186px;
}

.tr{
    height: 10px;
    width:185px;
}

.th{
    width: 125px;
}

.td{
    width: 60px;
}
