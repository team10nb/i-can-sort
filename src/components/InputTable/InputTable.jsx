/*
    Author: Yijie Lu, Shiliang Chen
*/

import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import PlayArrowIcon from '@material-ui/icons/PlayArrow';
import {color} from '../../scenes/mainPages/Correctness';
import {Tick, Cross} from '../TickCross/TickCross';
import Typography from '@material-ui/core/Typography';


const useStyles = makeStyles((theme) => ({
    buttons: {
        display:"flex",
        marginLeft: 28,
        marginTop: 20,
        marginBottom: 10,
    },
    title:{
        display:"flex",
        marginLeft: 35,
        fontFamily: "inherit",
        fontWeight: "700",
        fontSize:"16px"
    },
    div1:{
        display:"flex",
        marginTop: 0,
        '& > *': {
        margin: theme.spacing(1),
        },
        marginBottom:"19px"
      },
    
   
    button:{
        height:10
    }
    
    
    }));

function createData(str) {
    return {str};
}
      
export default function InputTable(props) {
    const classes = useStyles();

    const {
        legalShuffle,
        illegalShuffle,
        inputLegalString,
        inputIllegalString, 
        isCompleteLegal, 
        isCompleteIllegal, 
        legalClick, 
        illegalClick
    } = props;

    useEffect(() => {
        console.log(isCompleteIllegal)
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [isCompleteIllegal]);

    const legalRows = [
        createData(inputLegalString[0]),
        createData(inputLegalString[1]),
        createData(inputLegalString[2]),
    ]

    const illegalRows = [
        createData(inputIllegalString[0]),
        createData(inputIllegalString[1]),
        createData(inputIllegalString[2]),

    ]

    return(
        <div className = {classes.root}>
            <div>
                <div className = {classes.title}>  
                Legal Input
                </div>
                
                <div className = {classes.buttons}>                  
                    <Button 
                        variant="contained" 
                        style={{backgroundColor:color, color:"#FFFFFF", marginRight: 10, height: 30, width: 80}}
                        onClick = {legalShuffle}
                        
                    >
                        shuffle
                    </Button>
                    <IconButton 
                        variant="contained" 
                        size = 'small'
                        style={{backgroundColor:color, color:"#FFFFFF"}} 
                        onClick={legalClick}
                        
                    >
                        <PlayArrowIcon  />
                    </IconButton>                   
                </div>
                <div className = {classes.div1}>
                    <TableContainer component={Paper} >
                        <Table className={classes.table} size="small" aria-label="a dense table">
                            <TableHead>
                            <TableRow >
                                <TableCell >Input</TableCell>
                                <TableCell align="center" >Result</TableCell>
                            </TableRow>
                            </TableHead>
                            <TableBody>
                                
                                {legalRows.map((row, key) => (
                                    <TableRow key={key + "legal"} >
                                        <TableCell component="th" scope="row" >
                                            <div style={{width:"60px", fontWeight:"700"}}>{row.str}</div>
                                        </TableCell>
                                        <TableCell align="center"  > <Tick isComplete={isCompleteLegal[key]}/></TableCell>
                                    </TableRow>
                                ))}
                            </TableBody>
                        </Table>
                    </TableContainer>
                </div>
            </div>
            <Typography variant="overline" display="block" gutterBottom>
               
            </Typography>
            <div className = {classes.title}>
                Illegal Input
            </div>
            
            <div>
                <div className = {classes.buttons}>
                <Button 
                        variant="contained" 
                        style={{backgroundColor:color, color:"#FFFFFF", marginRight: 10, height: 30, width: 80}}
                        onClick = {illegalShuffle}
                    >
                        shuffle
                    </Button>
                    <IconButton 

                        variant="contained" 
                        size = 'small'
                        style={{backgroundColor:color, color:"#FFFFFF"}} 
                        onClick={illegalClick}
                        
                    >
                        <PlayArrowIcon  />
                    </IconButton>               
                </div>
                <div className = {classes.div1}>
                    <TableContainer component={Paper} >
                        <Table className={classes.table} size="small" aria-label="a dense table">
                            <TableHead>
                            <TableRow>
                                <TableCell>Input</TableCell>
                                <TableCell align="center">Result</TableCell>
                            </TableRow>
                            </TableHead>
                            <TableBody>
                                {illegalRows.map((row, key) => (
                                    <TableRow key={key + "illegal"} >
                                        <TableCell component="th" scope="row" >
                                        <div style={{width:"60px", fontWeight:"700"}}>{row.str}</div>
                                        </TableCell>
                                        <TableCell align="center" > <Cross isComplete={isCompleteIllegal[key]}/></TableCell>
                                    </TableRow>
                                ))}   
                            </TableBody>
                        </Table>
                    </TableContainer>
                </div>
            </div>
            
            
        </div>
    );
}
