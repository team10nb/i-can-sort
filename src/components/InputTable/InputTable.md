
### **Developer**

Shiliang Chen, Yijie Lu

Reference: MATERIAL-UI (https://material-ui.com)



###  **Introduction**

Input table is used for store the lagal input, illegal input, and their result. It is the component of InputTutorial. 

###  **Import**

```html
import React, {useEffect} from 'react';
import { makeStyles } from '@material-ui/core/styles';
```

###  **Props**


| Name | Type | Default | Description |
| ---- | ---- | ------- | ----------- |
|legalShuffle      |() => void|                                    |click shuffle can get legal input|
|illlegalShuffle   |() => void|                                    |click shuffle can get illlegal input|
|inputLegalString  |string[]  |["2,1,6,10","9,12,8,3","3,2,17,10",]|the lagal string table used |
|inputIllegalString|string[]  |["2,a,6,10","9,*,&,3","3,#,&,10",]  |the illagal string table used |
|legalArrays       |number[][]|["2,1,6,10","9,12,8,3","3,2,17,10",]|the lagal array animation should operate |
|isCompleteIllegal |boolean[] |[false,false,false,]                |the signal when should the cross svg be visible |
|isCompleteLegal   |boolean[] |[false,false,false,]                |the signal when should the tick svg be visible |
|legalClick        |() => void|                                    |click create should let the animation display |
|illegalClick      |() => void|                                    |click create should let the animation display |


###  **Methods**

| Name | Props | Return | Description |
| ---- | ----- | ------ | ----------- |
|createData|str|    string     |     get the string from the string[]      |
|InputTable|props|    a input table     |     a table store the lagal input, illegal input, and their result.      |

###  **Demo**

```jsx
<InputTable
    inputLegalString={["2,1,6,10","9,12,8,3","3,2,17,10",]}
    inputIllegalString={["2,a,6,10","9,*,&,3","3,#,&,10",]}
    legalArrays={["2,1,6,10","9,12,8,3","3,2,17,10",]}
    isCompleteIllegal={[false,false,false,]}
    isCompleteLegal={[false,false,false,]}
/>
```
