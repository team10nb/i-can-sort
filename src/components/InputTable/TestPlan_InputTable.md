## Test Plan 
--------
<br>

#### ComponentName: InputTable

|Test Name|Description|Operation|Expected Outcome|
|----|------|--------|----------------|
| should contain title & button |test whether titles and buttons are rendered properly| | Contain Legal Input, Illegal Input, Input and Result; Contain 4 buttons |
| should match InputTable Snapshot | Compare new snapshot with old one. |  | Snapshots are the same. |








