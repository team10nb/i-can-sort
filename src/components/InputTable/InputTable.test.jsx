/*
    Author: Shiliang Chen, Yijie Lu
*/
import InputTable from "./InputTable";
import { render } from "@testing-library/react";

jest.mock("../TickCross/TickCross",()=>{
    return {
        Tick: () => <div>mockTick</div>,
        Cross: () => <div>mockCross</div>,

    }
})


const props = {
    inputLegalString:["2,1,6,10","9,12,8,3","3,2,17,10",],
    inputIllegalString:["2,a,6,10","9,*,&,3","3,#,&,10",],
    legalArrays:["2,1,6,10","9,12,8,3","3,2,17,10",],
    isCompleteIllegal:[false,false,false,],
    isCompleteLegal:[false,false,false],
}


test( 'should contain title & button', () => {
    const content = render(<InputTable {...props}  />)
    const {getByText, getAllByText, getAllByRole} = content;
    getByText("Legal Input");
    getByText("Illegal Input");

    getAllByText("Input");
    getAllByText("Result");

    const buttons = getAllByRole("button");
    expect(buttons.length).toBe(4);

})

test( 'should match InputTable snapshot', () => {
    const content = render(<InputTable {...props}  />);
    expect(content).toMatchSnapshot();

})
