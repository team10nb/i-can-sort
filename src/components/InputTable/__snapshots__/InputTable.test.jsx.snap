// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match InputTable snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-6"
      >
        <div>
          <div
            class="makeStyles-title-8"
          >
            Legal Input
          </div>
          <div
            class="makeStyles-buttons-7"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="0"
              title="shuffle"
              type="button"
              variant="contained"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="0"
              title="play"
              type="button"
              variant="contained"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8 5v14l11-7z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-tables-9"
          >
            <div
              class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <table
                aria-label="a dense table"
                class="MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                      scope="col"
                    >
                      Input
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                      scope="col"
                    >
                      Result
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        2,1,6,10
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockTick
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        9,12,8,3
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockTick
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        3,2,17,10
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockTick
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <span
          class="MuiTypography-root MuiTypography-overline MuiTypography-gutterBottom MuiTypography-displayBlock"
        />
        <div
          class="makeStyles-title-8"
        >
          Illegal Input
        </div>
        <div>
          <div
            class="makeStyles-buttons-7"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="0"
              title="shuffle"
              type="button"
              variant="contained"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
              style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
              tabindex="0"
              title="play"
              type="button"
              variant="contained"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8 5v14l11-7z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-tables-9"
          >
            <div
              class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <table
                aria-label="a dense table"
                class="MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                      scope="col"
                    >
                      Input
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                      scope="col"
                    >
                      Result
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        2,a,6,10
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockCross
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        9,*,&,3
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockCross
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                      <div
                        style="width: 60px; font-weight: 700;"
                      >
                        3,#,&,10
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    >
                       
                      <div>
                        mockCross
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-6"
    >
      <div>
        <div
          class="makeStyles-title-8"
        >
          Legal Input
        </div>
        <div
          class="makeStyles-buttons-7"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="0"
            title="shuffle"
            type="button"
            variant="contained"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="0"
            title="play"
            type="button"
            variant="contained"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-tables-9"
        >
          <div
            class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="a dense table"
              class="MuiTable-root"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    scope="col"
                  >
                    Input
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    scope="col"
                  >
                    Result
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      2,1,6,10
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockTick
                    </div>
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      9,12,8,3
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockTick
                    </div>
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      3,2,17,10
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockTick
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-overline MuiTypography-gutterBottom MuiTypography-displayBlock"
      />
      <div
        class="makeStyles-title-8"
      >
        Illegal Input
      </div>
      <div>
        <div
          class="makeStyles-buttons-7"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="0"
            title="shuffle"
            type="button"
            variant="contained"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
            style="background-color: rgb(255, 111, 0); color: rgb(255, 255, 255);"
            tabindex="0"
            title="play"
            type="button"
            variant="contained"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8 5v14l11-7z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-tables-9"
        >
          <div
            class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="a dense table"
              class="MuiTable-root"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    scope="col"
                  >
                    Input
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                    scope="col"
                  >
                    Result
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      2,a,6,10
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockCross
                    </div>
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      9,*,&,3
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockCross
                    </div>
                  </td>
                </tr>
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                    <div
                      style="width: 60px; font-weight: 700;"
                    >
                      3,#,&,10
                    </div>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeSmall"
                  >
                     
                    <div>
                      mockCross
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
